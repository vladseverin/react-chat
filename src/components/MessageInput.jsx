import React from 'react';
import { withStyles } from 'material-ui/styles';
import Input from 'material-ui/Input';
import Paper from 'material-ui/Paper';

const style = theme => ({
  messageInputWrapper: {
    position: 'fixed',
    left: 'auto',
    right: 0,
    bottom: 0,
    width: 'calc(100% - 320px)',
    padding: theme.spacing.unit * 3,
  },
  messageInput: {
    padding: theme.spacing.unit * 3,
  },
});

const MessageInput = ({ classes }) => (
  <div className={classes.messageInputWrapper}>
    <Paper className={classes.messageInput}>
      <Input
        placeholder="Type your message..."
        fullWidth
      />
    </Paper>
  </div>
);

export default withStyles(style)(MessageInput);
