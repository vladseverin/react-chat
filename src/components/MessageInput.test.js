import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from './MessageInput';

const mockProps = {
  showJoinButton: false,
  onJoinButtonClick: jest.fn(),
  disabled: false,
  sendMessage: jest.fn(),
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
