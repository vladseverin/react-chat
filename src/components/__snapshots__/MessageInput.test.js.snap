// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders correctly 1`] = `
<div
  className="MessageInput-messageInputWrapper-43"
>
  <div
    className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 MessageInput-messageInput-44"
  >
    <div
      className="MuiInput-root-72 MuiInput-fullWidth-79 MuiInput-underline-76"
      onKeyPress={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-80"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your messageâ€¦"
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
